File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.scala
import scala.collection.mutable.HashSet
object Solution {
    def containsNearbyDuplicate(nums: Array[Int], k: Int): Boolean = {
        val n = nums.length
        val hs = HashSet[Int]()
        var i = 0
        while(i < n){
            if (hs.contains(nums(i))){
                return true
            } else {
                hs += nums(i)
            }
            if (hs.size > k){
                hs -= nums(i-k)
            }
            i += 1
        }
        return false
    }
}
//TC, SC: O(n)

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        hs = set()
        for i in range(n):
            if nums[i] in hs:
                return True
            else:
                hs.add(nums[i])
            if len(hs) > k:
                hs.remove(nums[i-k])
        return False

# TC, SC: O(n)

        

# Synced seamlessly with LeetHub Pro
# Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
# Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna